pkg/networkpolicy/controller.go

```diff
@@ -681,83 +682,97 @@ func (c *Controller) syncNFTablesRules(ctx context.Context) error {
 	// IPVS packets follow a different path in netfilter, so we process
 	// everything in the POSTROUTING hook before SNAT happens.
 	// Ref: https://github.com/kubernetes-sigs/kube-network-policies/issues/46
-	for _, hook := range []knftables.BaseChainHook{knftables.PostroutingHook} {
-		chainName := string(hook)
-		tx.Add(&knftables.Chain{
-			Name:     chainName,
-			Type:     knftables.PtrTo(knftables.FilterType),
-			Hook:     knftables.PtrTo(hook),
-			Priority: knftables.PtrTo(knftables.SNATPriority + "-5"),
-		})
-		tx.Flush(&knftables.Chain{
-			Name: chainName,
+	hook := knftables.PostroutingHook
+	chainName := string(hook)
+	tx.Add(&knftables.Chain{
+		Name:     chainName,
+		Type:     knftables.PtrTo(knftables.FilterType),
+		Hook:     knftables.PtrTo(hook),
+		Priority: knftables.PtrTo(knftables.SNATPriority + "-5"),
+	})
+	tx.Flush(&knftables.Chain{
+		Name: chainName,
+	})
+
+	// DNS is processed by addDNSRacersWorkaroundRules()
+	// TODO: remove once kernel fix is on most distros
+	if c.config.NetfilterBug1766Fix {
+		tx.Add(&knftables.Rule{
+			Chain:   chainName,
+			Rule:    "udp dport 53 accept",
+			Comment: ptr.To("process DNS traffic on PREROUTING hook with network policy enforcement to avoid netfiler race bug"),
```

Comment:
```suggestion
			Comment: ptr.To("process DNS traffic on PREROUTING hook with network policy enforcement to avoid netfilter race bug"),
```
or

```suggestion
			Comment: ptr.To("process DNS traffic on PREROUTING hook with network policy enforcement to avoid netfilter race condition bug"),
```